src/qt/bitcoin.cpp

```diff
@@ -465,8 +471,17 @@ void BitcoinApplication::initializeResult(bool success)
     qDebug() << __func__ << ": Initialization result: " << success;
     // Set exit result.
     returnValue = success ? EXIT_SUCCESS : EXIT_FAILURE;
-    if(success)
-    {
+    if (!success) {
+        try {
+            m_node.appShutdown();
```

Comment:
I don't think it is right to call `appShutdown()` here because it should be covered by `return quit();` below. `quit` should cause the first application loop to quit, which should trigger `requestShutdown()`:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L718-L720

which should trigger `requestedShutdown()`:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L404-L431

which should trigger `BitcoinCore::shutdown`:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L379

which already calls `appShutdown()`:

https://github.com/bitcoin/bitcoin/blob/59ecacfc84af13e5a1608e7d970315d07dcb0269/src/qt/bitcoin.cpp#L268-L281

The shutdown process is definitely overcomplicated and should be simplified. I think a good direction would be to stop calling [`QApplication::quit`](http://doc.qt.io/qt-5/qcoreapplication.html#quit) during shutdown and instead just call it after. So Qt shutdown would just immediately follow bitcoin shutdown instead of being mingled with it.